connectionmanager/connectionmanager.go

```diff
@@ -105,6 +106,10 @@ func connect(
 				log.Errorf("connection manager failed to connect to %s%s: %s; "+
 					"reconnecting in %ds (try %d/%d); len(token)=%d", serverUrl, connectUrl,
 					err.Error(), reconnectIntervalSeconds, i, retries, len(token))
+				if strings.Contains(err.Error(), "malformed ws or wss URL") {
+					// If the given serverUrl is malformed, exit so that the caller can update it
+					return err
+				}
```

Comment:
this on I understand, but I believe we should not enter the connect state if the serverURl is empty, surprisingly `url.Parse("")` return nil error, so we have to check it manually. I would suggest waiting for a non empty ServerURL (I could swear we were/are doing it)

Reply:
I will look more into it. I believe we have code for it when parsing the D-Bus response but we must have some bug somewhere.